crates/autopilot/src/infra/persistence/mod.rs

```diff
@@ -382,6 +388,156 @@ impl Persistence {
         Ok(solution)
     }
 
+    /// Computes solvable orders based on the latest observed block number,
+    /// order creation timestamp, and minimum validity period.
+    pub async fn solvable_order_after(
+        &self,
+        current_orders: HashMap<domain::OrderUid, model::order::Order>,
+        after_timestamp: DateTime<Utc>,
+        after_block: u64,
+        min_valid_to: u32,
+    ) -> anyhow::Result<boundary::SolvableOrders> {
+        let after_block = i64::try_from(after_block).context("block number value exceeds i64")?;
+        let mut tx = self.postgres.pool.begin().await.context("begin")?;
+        // Set the transaction isolation level to REPEATABLE READ
+        // so all the SELECT queries below are executed in the same database snapshot
+        // taken at the moment before the first query is executed.
+        sqlx::query("SET TRANSACTION ISOLATION LEVEL REPEATABLE READ")
+            .execute(tx.deref_mut())
+            .await?;
+
+        // Find order uids for orders that were updated after the given block.
+        let updated_order_uids: Vec<database::OrderUid> = {
+            let _timer = Metrics::get()
+                .database_queries
+                .with_label_values(&["updated_order_uids"])
+                .start_timer();
+
+            database::orders::updated_order_uids_after(&mut tx, after_block)
+                .map_ok(|uid| uid.0)
+                .try_collect()
+                .await?
+        };
+
+        // Fetch the orders that were updated after the given block and were created or
+        // cancelled after the given timestamp.
+        let next_orders: HashMap<domain::OrderUid, model::order::Order> = {
+            let _timer = Metrics::get()
+                .database_queries
+                .with_label_values(&["open_orders_after"])
+                .start_timer();
+
+            database::orders::open_orders_by_time_and_uids(
+                &mut tx,
+                &updated_order_uids,
+                after_timestamp,
+            )
+            .map(|result| match result {
+                Ok(order) => full_order_into_model_order(order)
+                    .map(|order| (domain::OrderUid(order.metadata.uid.0), order)),
+                Err(err) => Err(anyhow::Error::from(err)),
+            })
+            .try_collect()
+            .await?
+        };
+
+        // Fetch quotes for new orders and also update them for the cached ones since
+        // they could also be updated.
```

Comment:
This is also needed because ethflow orders could theoretically be reorged?
We should really find a way to implement ethflow orders in a nicer way to remove all those ugly edge cases. :/

Reply:
> This is also needed because ethflow orders could theoretically be reorged?

Because of this code:
https://github.com/cowprotocol/services/blob/e7de6bfd0526e00b55fda817b33a0fde715c8a61/crates/autopilot/src/database/onchain_order_events.rs#L329-L342